src/sentry/south_migrations/0342_projectplatform.py

```diff
@@ -0,0 +1,3271 @@
+# -*- coding: utf-8 -*-
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import DataMigration
+from django.db import models
+from django.db.models import Count
+
+
+class Migration(DataMigration):
+    def forwards(self, orm):
+        db.commit_transaction()
+        try:
+            self._forwards(orm)
+        except Exception:
+            db.start_transaction()
+            raise
+        db.start_transaction()
+
+    def _forwards(self, orm):
+
+        Project = orm['sentry.Project']
+        ProjectPlatform = orm['sentry.ProjectPlatform']
+
+        # RangeQuerySetWrapperWithProgressBar throws a `AttributeError: 'dict' object has no attribute 'pk'`
+        # if I try to wrap the queryset with it.
+        # It's a relatively small dataset: the queryset will return about 70,000 rows in production.
+        queryset = ProjectPlatform.objects.values('project_id').annotate(
+            platforms=Count('platform')
+        )
+        for row in queryset:
+            if row['platforms'] == 1:
+                projectplatform = ProjectPlatform.objects.filter(project_id=row['project_id']
+                                                                 ).first()
+            else:
+                projectplatform = ProjectPlatform.objects.filter(project_id=row['project_id']
```

Comment:
Are there any cases where this would be an index error? Is it not possible for `row['platforms'] == 0` ?